internal/pkg/cli/svc_init.go

```diff
@@ -432,19 +428,22 @@ func parseHealthCheck(df dockerfileParser) (*manifest.ContainerHealthCheck, erro
 	}, nil
 }
 
-func dockerPlatform(engine dockerEngine, image string) (os, arch string, err error) {
-	os, arch = runtime.GOOS, runtime.GOARCH
-	if image == "" {
-		os, arch, err = engine.GetPlatform()
-		if err != nil {
-			return "", "", fmt.Errorf("get os/arch from docker: %w", err)
-		}
+func redirectPlatform(engine dockerEngine, image string) (string, error) {
+	// If the user passes in an image, their docker engine isn't necessarily running, and we can't redirect the platform because we're not building the Docker image.
+	if image != "" {
+		return "", nil
 	}
-	// Until we target X86_64 for ARM architectures, log a warning.
-	if arch == exec.ArmArch || arch == exec.Arm64Arch {
-		log.Warningf("Architecture type %s is currently unsupported.\nTo deploy, run %s\n", arch, "`DOCKER_DEFAULT_PLATFORM=linux/amd64 copilot deploy`")
+	_, arch, err := engine.GetPlatform()
+	if err != nil {
+		return "", fmt.Errorf("get os/arch from docker: %w", err)
 	}
-	return os, arch, nil
+	// Log a message informing non-default arch users of platform for build.
+	if arch != exec.Amd64Arch {
+		log.Warningf("Architecture type %s is currently unsupported. Setting platform %s instead.\n", arch, fmt.Sprintf(exec.FmtOSArch, exec.LinuxOS, exec.Amd64Arch))
```

Comment:
We can probably also direct people's attention to the manifest's `platform` field in this case!